### PR TITLE
James 1888 Tests for MessageIdManager

### PR DESCRIPTION
**Pull request by chibenwa**
Review only commits named JAMES-1888

Fixes : 
 - Messages should be able to live in several mailboxes
 - Do not return messages when mailbox do not exist
 - Guice injection
 - other small corrections
 - Manager for MessageId tests

Security will be handled in an other ticket

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java [hunk 1]
```diff
@@ -51,7 +51,6 @@
 import javax.mail.Flags.Flag;
 
 import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
-import org.apache.james.backends.cassandra.utils.CassandraConstants;
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java]: great, but you don't need to yell in the commit title ;)

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java [hunk 2]
```diff
@@ -182,16 +183,22 @@ private ByteBuffer toByteBuffer(InputStream stream) throws IOException {
 
     public CompletableFuture<Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>>> retrieveMessages(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, Optional<Integer> limit) {
         return retrieveRows(messageIds, fetchType, limit)
-                .thenApply(resultSet -> CassandraUtils.convertToStream(resultSet)
-                        .map(row -> message(row, messageIds, fetchType)));
+                .thenApply(resultSet -> toto(messageIds, fetchType, resultSet));
+    }
+
+    private Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> toto(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, ResultSet resultSet) {
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: toto?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 3]
```diff
@@ -206,15 +206,14 @@ private void addMessageToMailboxes(MessageIdMapper messageIdMapper, MailboxMessa
...
+            MessageMetaData metaData = save(mailboxSession, messageIdMapper, copy);
             dispatcher.added(mailboxSession, metaData, mailboxMapper.findMailboxById(mailboxId));
         }
     }
 
-    private MessageMetaData save(MailboxSession mailboxSession, MessageIdMapper messageIdMapper, MailboxId mailboxId, MailboxMessage mailboxMessage) throws MailboxException {
-        validateQuota(ImmutableList.of(mailboxId), ImmutableList.<MailboxId>of(), mailboxSession, mailboxMessage);
-        long modSeq = mailboxSessionMapperFactory.getModSeqProvider().nextModSeq(mailboxSession, mailboxId);
-        MessageUid uid = mailboxSessionMapperFactory.getUidProvider().nextUid(mailboxSession, mailboxId);
+    private MessageMetaData save(MailboxSession mailboxSession, MessageIdMapper messageIdMapper, MailboxMessage mailboxMessage) throws MailboxException {
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: where do you validate quota then?

Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: Quota is validated in setInMailboxes command (upfront) as we need to group by quotaRoot to allow move and deletion even when overQuota.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java (hunk 2 above)
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: Ahhhhh oups

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 3 above)
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: that's fine, thanks for the explanation

**Comment by chibenwa** [comment]
test this please

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java [hunk 4]
```diff
@@ -183,10 +183,10 @@ private ByteBuffer toByteBuffer(InputStream stream) throws IOException {
 
     public CompletableFuture<Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>>> retrieveMessages(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, Optional<Integer> limit) {
         return retrieveRows(messageIds, fetchType, limit)
-                .thenApply(resultSet -> toto(messageIds, fetchType, resultSet));
+                .thenApply(resultSet -> toMessagesWithoutAttachements(messageIds, fetchType, resultSet));
     }
 
-    private Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> toto(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, ResultSet resultSet) {
+    private Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> toMessagesWithoutAttachements(List<ComposedMessageIdWithMetaData> messageIds, FetchType fetchType, ResultSet resultSet) {
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: I don't understand the function name: you tell "withoutAttachment" but you are returning a pair where the second element is a Stream of MessageAttachmentRepresentation. It is expected?

**Pull request merged**